From:	Tomas Winkler <tomas.winkler@intel.com>
Subject: [3.10][PATCH 2/4] mei: don't have to clean the state on power up
Date:	Sun, 25 Aug 2013 12:49:47 +0300
Message-Id: <1377424189-5508-3-git-send-email-tomas.winkler@intel.com>

stable: 3.10
commit 99f22c4ef24cf87b0dae6aabe6b5e620b62961d9 upstream

When powering up, we don't have to clean up the device state
nothing is connected.

Cc: stable@vger.kernel.org
Tested-by: Shuah Khan <shuah.kh@samsung.com>
Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/misc/mei/init.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/misc/mei/init.c
+++ b/drivers/misc/mei/init.c
@@ -143,7 +143,8 @@ void mei_reset(struct mei_device *dev, i
 
 	dev->hbm_state = MEI_HBM_IDLE;
 
-	if (dev->dev_state != MEI_DEV_INITIALIZING) {
+	if (dev->dev_state != MEI_DEV_INITIALIZING &&
+	    dev->dev_state != MEI_DEV_POWER_UP) {
 		if (dev->dev_state != MEI_DEV_DISABLED &&
 		    dev->dev_state != MEI_DEV_POWER_DOWN)
 			dev->dev_state = MEI_DEV_RESETTING;
